src/test/java/org/beta/tchap/identite/matrix/rest/MatrixBotIntTest.java

```diff
@@ -0,0 +1,99 @@
+package org.beta.tchap.identite.matrix.rest;
+
+import org.apache.log4j.BasicConfigurator;
+import org.beta.tchap.identite.bot.BotSender;
+import org.beta.tchap.identite.matrix.rest.homeserver.HomeServerService;
+import org.beta.tchap.identite.matrix.rest.login.LoginService;
+import org.beta.tchap.identite.matrix.rest.room.DirectRoomsResource;
+import org.beta.tchap.identite.matrix.rest.room.RoomClient;
+import org.beta.tchap.identite.matrix.rest.room.RoomClientFactory;
+import org.beta.tchap.identite.matrix.rest.room.RoomService;
+import org.beta.tchap.identite.utils.Constants;
+import org.beta.tchap.identite.utils.Environment;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.beta.tchap.identite.matrix.rest.homeserver.HomeServerService.buildHomeServerUrl;
+
+
+// FIXME need cleanup of rooms before/after each test
+class MatrixBotIntTest {
+    private static RoomService roomService;
+    private static BotSender botSender;
+
+    @BeforeAll
+    public static void setup() {
+        // Needed for logging
+        BasicConfigurator.configure();
+
+        String account = Environment.getenv(Constants.TCHAP_ACCOUNT);
+        String password = Environment.getenv(Constants.TCHAP_PASSWORD);
+
+        HomeServerService homeServerService = new HomeServerService();
+        LoginService loginService = new LoginService();
+
+        String accountHomeServerUrl = buildHomeServerUrl(homeServerService.findHomeServerByEmail(account));
+        String accessToken = loginService.findAccessToken(accountHomeServerUrl, account, password);
+
+        RoomClient roomClient = RoomClientFactory.build(accountHomeServerUrl, accessToken);
+        roomService = new RoomService(roomClient);
+
+        botSender = new BotSender(roomService);
+    }
+
+    @Nested
+    class NoEventsTest {
+        @Test
+        void shouldHaveNoDMEventsIfNoDM() {
+            String destId = "@calev.eliacheff-beta.gouv.fr:i.tchap.gouv.fr";
+
+            DirectRoomsResource dmRooms = roomService.listDMRooms();
+            Assertions.assertNull(dmRooms.getDirectRoomsForMId(destId));
+        }
+    }
+
+    @Nested
+    class CreateDMTest {
+        @Test
+        void shouldCreateADMAndAddDMEvent() {
+            String destId = "@calev.eliacheff-beta.gouv.fr:i.tchap.gouv.fr";
```

Comment:
we should use a user for testing and remove your id from the code

Reply:
Yeah, I would need some guidance about the best way to generify the code